http_parser.c

```diff
@@ -472,6 +483,85 @@ static struct {
 
 int http_message_needs_eof(const http_parser *parser);
 
+const char* find_crlf(const char* p, const char* data, size_t len);
+
+#if defined(_MSC_VER) // SSE2 is baseline
+#include <intrin.h>
+#define __builtin_ctz _tzcnt_u32
+#elif defined(__SSE2__) && defined(__GNUC__)
+#include <emmintrin.h>
+#endif
+
+#if defined(_MSC_VER) || (defined(__SSE2__) && defined(__GNUC__))
+
+int32_t get_crlf_mask(const char* p) {
+  /* [ c, 0, 0, 0, 0, 0 .. 0 ] */
+  const __m128i vCR = _mm_set1_epi8(0x0a);
+
+  /* [ c, 0, 0, 0, 0, 0 .. 0 ] */
+  const __m128i vLF = _mm_set1_epi8(0x0d);
+
+  __m128i v1, v2;
+  v1 = *(__m128i*)p;
+  v2 = _mm_cmpeq_epi8(vCR, v1);
+  v1 = _mm_cmpeq_epi8(vLF, v1);
+  v2 = _mm_or_si128(v1, v2);
+  return _mm_movemask_epi8(v2);
+}
+
+const char* find_crlf(const char* p, const char* data, size_t len) {
+  const char* lastp = MIN(data + len, HTTP_MAX_HEADER_SIZE + p);
+
+  int32_t result = 0;
+
+  size_t alignment = (long) p & 15;
+  p -= alignment;
+
+  result = get_crlf_mask(p + 16) << 16 | get_crlf_mask(p);
+  result = (result >> alignment) << alignment;
+
+  if (!result) {
+    while(!result && lastp >= p+32) {
```

Comment:
Inconsistent spacing. Personally I prefer a space between the keyword and parenthesis.

Reply:
done